xonsh/jobs.py

```diff
@@ -6,13 +6,26 @@
 import signal
 import builtins
 from subprocess import TimeoutExpired, check_output
-from io import BytesIO
 from collections import deque
 
-from xonsh.tools import ON_WINDOWS
+from xonsh.platform import ON_DARWIN, ON_WINDOWS
 
 tasks = deque()
 
+
+if ON_DARWIN:
+    def _send_signal(job, signal):
+        # On OS X, os.killpg() may cause PermissionError when there are
+        # any zombie processes in the process group.
+        # See github issue #1012 for details
+        for pid in job['pids']:
+            os.kill(pid, signal)
+
+else:
+    def _send_signal(job, signal):
```

Comment:
Good catch. I have no idea if this is a problem. I think @BYK has a better grasp on these things. I will try to run with it tomorrow to see if it matters.